### PR TITLE
feat: click-through analytics

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Implements click-through rate analytics as per the [technical spec](https://www.notion.so/walletconnect/Notify-API-click-through-tracking-tech-doc-51748e76c53644e995909f15fc615f3a?pvs=4).

## How Has This Been Tested?

New tests

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/analytics/follow_notification_link.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    crate::model::types::AccountId,
+    parquet_derive::ParquetRecordWriter,
+    relay_rpc::domain::{ProjectId, Topic},
+    serde::Serialize,
+    std::sync::Arc,
+    uuid::Uuid,
+};
+
+pub struct FollowNotificationLinkParams {
+    pub project_pk: Uuid,
```
Comment (dnul) [inline src/analytics/follow_notification_link.rs:11]: this looks good, would add geo-data and user-agent data ( e.g OS / browser ) stuff. 

We'd probably like to see in the future CTR by different cohorts and whatnot.. for mvp this is awesome

Comment (chris13524) [reply src/analytics/follow_notification_link.rs]: Added!

File: src/state.rs [hunk 2]
```diff
@@ -96,10 +100,10 @@ impl AppState {
         event: WebhookNotificationEvent,
         account: &str,
     ) -> Result<(), NotifyServerError> {
-        if !is_valid_account(account) {
-            info!("Didn't register account - invalid account: {account} for project {project_id}");
-            return Err(NotifyServerError::InvalidAccount);
-        }
+        // if !is_valid_account(account) {
```
Comment (geekbrother) [inline src/state.rs:103]: What's the reason this commented out?

**Review by geekbrother — APPROVED** [review]
lgtm

File: src/state.rs (hunk 2 above)
Comment (chris13524) [reply src/state.rs:103]: Unrelated, I was getting IDE errors due to switching between branches on the `lazy_static` usage so I decided to just remove the dependency. It was only used by `is_valid_account()` and that's only used here but the webhook stuff is not in-use for now so decided to just delete/comment it out. Will replace with the `AccountId` type in the future when/if we re-introduce webhooks.

**Pull request merged**